test/apps/iodemo/io_demo.cc

```diff
@@ -526,11 +526,23 @@ class DemoClient : public P2pDemoCommon {
         return !_error_flag;
     }
 
-    UcxConnection* connect(const char* server_addr) {
+    UcxConnection* connect(const char* server) {
         struct sockaddr_in connect_addr;
+        char str[64];
+        const char* server_addr;
+        int   port_num;
+
         memset(&connect_addr, 0, sizeof(connect_addr));
         connect_addr.sin_family = AF_INET;
-        connect_addr.sin_port   = htons(opts().port_num);
+        if (strchr(server, ':') == NULL) {
+            port_num = opts().port_num;
+            server_addr = server;
```

Comment:
basically LGTM, I would suggest just minor usability improvement:
does it mean that command line could be:
io_demo [options] **port0 port1**... ? I guess it assumes if servers are running on the same host
maybe to rework it as follow:
io_demo [options] **:port0 :port1**...
to make it clearer
IMHO this way the code could look simpler as well - no need second branch
@yosefe WDYT?

Reply:
ahh.. I see, it handles 
`io_demo [options] -p port addr1 addr2`
and
`io_demo [options] -p port0 addr0 addr1:port addr2:port2`

Reply:
> ahh.. I see, it handles
> `io_demo [options] -p port addr1 addr2`
> and
> `io_demo [options] -p port0 addr0 addr1:port addr2:port2`

@evgeny-leksikov Yes, you are right.